### PR TITLE
feat: delete artwork flow in edit artwork

### PR DESCRIPTION
**Pull request by dariakoko**
The type of this PR is: **Feat**

This PR solves [CX-2757]

### Description

Added delete artwork flow in edit artwork

| Web | Mobile |
| --- | --- |
| image | image |

+ corrected the layout of the modal -- removed extra margin
image

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CX-2757]: https://artsyproduct.atlassian.net/browse/CX-2757?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx [hunk 1]
```diff
@@ -106,6 +112,92 @@ export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = (
...
+          : "Your artwork will not be added to My Collection."}
+      </Text>
+    </ModalDialog>
+  )
+
+  const handleDelete = async () => {
+    try {
+      await deleteArtworkRequest({
+        artworkId: artwork?.internalID!,
+      }).then(() => router.push({ pathname: "/settings/my-collection" }))
```
Comment (olerichter00) [inline src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx:158]: **(suggestion):** If you're already using `await` here, you could get rid of the callback function like this:

```suggestion
      await deleteArtworkRequest({
        artworkId: artwork?.internalID!,
      })
      
      router.push({ pathname: "/settings/my-collection" })
```

File: src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx [hunk 2]
```diff
@@ -106,6 +112,92 @@ export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = (
...
+        variant: "error",
+        message: "An error occurred",
+        description: "Please contact support@artsymail.com",
+      })
+
+      return
+    }
+  }
+
+  const deletionModal = (
```
Comment (olerichter00) [inline src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx:172]: What do you think about moving both Modals into their own files? This file is getting pretty long....

File: src/Apps/MyCollection/Routes/EditArtwork/Utils/useCreateOrUpdateOrDeleteArtwork.ts [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
 import { useCreateArtwork } from "../Mutations/useCreateArtwork"
 import { useUpdateArtwork } from "../Mutations/useUpdateArtwork"
+import { useDeleteArtwork } from "../Mutations/useDeleteArtwork"
 
 export type ArtworkInput =
   | MyCollectionCreateArtworkInput
   | MyCollectionUpdateArtworkInput
 
-export const useCreateOrUpdateArtwork = () => {
+export const useCreateOrUpdateOrDeleteArtwork = () => {
```
Comment (olerichter00) [inline src/Apps/MyCollection/Routes/EditArtwork/Utils/useCreateOrUpdateOrDeleteArtwork.ts:12]: I wonder if it makes sense to do this change here. We could use `const { submitMutation: deleteArtwork } = useDeleteArtwork()` directly in [MyCollectionArtworkForm.tsx](https://github.com/artsy/force/pull/10834/files#diff-7445b7d2b9d4df4ded453cf7511b68c1234ad7d6168cabd72b81d75309fa2cce)`. I've just added this helper to upudate-or-create an artwork with one function call.

**Review by olerichter00 — CHANGES_REQUESTED** [review]
This looks great! I've added some suggestions and comments.

File: src/Apps/MyCollection/Routes/EditArtwork/Utils/useCreateOrUpdateOrDeleteArtwork.ts (hunk 3 above)
Comment (dariakoko) [reply src/Apps/MyCollection/Routes/EditArtwork/Utils/useCreateOrUpdateOrDeleteArtwork.ts:12]: I don't like the idea of having requests in separate files. I think it makes sense to have everything in one place
What if I keep upudate-or-create function and add another one only for deletion to the same file?

File: src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx (hunk 2 above)
Comment (dariakoko) [reply src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx:172]: Agree :)

File: src/Apps/MyCollection/Routes/EditArtwork/Utils/useCreateOrUpdateOrDeleteArtwork.ts (hunk 3 above)
Comment (olerichter00) [reply src/Apps/MyCollection/Routes/EditArtwork/Utils/useCreateOrUpdateOrDeleteArtwork.ts:12]: It seems to be the preferred pattern in Force to use the `useMutation` pattern and than use `commitMutation` directly in the component that uses the file ([see comment from @damassi ](https://github.com/artsy/force/pull/10732#discussion_r942810460)).

Comment (dariakoko) [reply src/Apps/MyCollection/Routes/EditArtwork/Utils/useCreateOrUpdateOrDeleteArtwork.ts:12]: Aaaah, alright, I didn't know
Changing to the preferred way

Comment (olerichter00) [reply src/Apps/MyCollection/Routes/EditArtwork/Utils/useCreateOrUpdateOrDeleteArtwork.ts:12]: Thanks 🙇

**Review by olerichter00 — APPROVED** [review]

**Review by olerichter00 — APPROVED** [review]

**Pull request merged**